### PR TITLE
Implement tier selector on product page

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

Implement selecting tier when we get multiple product offers from backend.

Refactor the components to better match the final wireframes.

## Justify why they are needed

We should revisit this setup soon but didn't want to create an even larger PR.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

[GRW-1701]: https://hedvig.atlassian.net/browse/GRW-1701?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #762** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/762" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/762?utm_source=stack-comment).

File: apps/store/src/components/ProductPage/TierSelector/TierSelector.tsx [hunk 1]
```diff
@@ -31,29 +32,23 @@ export const TierSelector = ({ product, cartId, priceIntent, onAddedToCart }: Pr
...
-      </div>
+      <PriceCardForm
+        onSubmit={handleSubmitAddToCart}
+        loading={loadingAddToCart}
+        offers={priceIntent.offers}
+        selectedOffer={productOffer}
+        onChangeOffer={onChangeOffer}
+      />
 
       <PriceCalculatorFooterForm
```
Comment (gustaveen) [inline apps/store/src/components/ProductPage/TierSelector/TierSelector.tsx:45]: When do we show this component? :) I thought it was a sticky price button but can't see it on the page?

**Review by gustaveen — APPROVED** [review]
Nice job! 🙌 Tested it and it for works for car 👍
As you said, this looks different in the prototype, might be a dropdown instead. 
I'll check with William/Zak if we should change that now when I update the UI for Price Calculator 🙌

**Review by alebedev — APPROVED** [review]
Looks good

File: apps/store/src/components/ProductPage/TierSelector/TierSelector.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/components/ProductPage/TierSelector/TierSelector.tsx:45]: Yeah I noticed it had disappeared, not sure why...

Also don't see it in the updated designs so maybe talk to them first - maybe we just remove it

**Pull request merged**